doc/releases/changelog-dev.md

```diff
@@ -6,6 +6,9 @@
 
 <h3>Improvements 🛠</h3>
 
+* `DefaultQubit2` devices have a `_max_workers` attribute which controls the process pool size for batch execution.
```

Comment:
Happy to leave as is for now, but hopefully DQ2 will become DQ by the release, so we can rephrase this entry a little.